### PR TITLE
fix: avoid invalid identifier error when validate genesis

### PR DESCRIPTION
**Pull request by mmsqe**
that contains localhost client

## Description

closes: https://github.com/cosmos/ibc-go/issues/7396

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/03-connection/types/keys.go [hunk 1]
```diff
@@ -52,6 +53,9 @@ func IsValidConnectionID(connectionID string) bool {
 
 // ParseConnectionSequence parses the connection sequence from the connection identifier.
 func ParseConnectionSequence(connectionID string) (uint64, error) {
+	if connectionID == exported.LocalhostConnectionID {
```
Comment (colin-axner) [inline modules/core/03-connection/types/keys.go:56]: My only hesitation with this fix is that `ParseConnectionSequence("connection-0")` and `ParseConnectionSequence("connection-localhost")` would return the same result

I wonder if we should maintain that calling `ParseConnectionSequence` on the localhost connection is an invalid function call and instead move the if check on the localhost connection id to `IsValidConnectionID` and the genesis.go validation function?

**Review by colin-axner — COMMENTED** [review]
Thank you @mmsqe for reporting the issue and providing a fix. I left a comment on an alternative solution, let me know what you think

**Review by colin-axner — APPROVED** [review]
Thanks @mmsqe for the followup! Everything looks good to me

**Review by damiannolan — APPROVED** [review]
Thank you @mmsqe!

**Comment by damiannolan** [comment]
Should we backport this?

**Comment by mmsqe** [comment]
> Should we backport this?

Would be nice if we can backport to v8.5.x.

**Pull request merged**